test/test_get_soup.py

```diff
@@ -0,0 +1,40 @@
+import requests
+import mock
+import pytest
+import sys
+
+sys.modules["config"] = mock.MagicMock()
+from amz_review_scraper.get_soup import boil_soup
+
+
+def mocked_requests_get(*args, **kwargs):
+    class MockResponse:
+        def __init__(self, json_data, text, status_code):
+            self.json_data = json_data
+            self.text = text
+            self.status_code = status_code
+
+        def json(self):
+            return self.json_data
+
+    print("args: " + args[0])
+    if args[0] == "https://www.amazon.com/gp/product/B111111111":
+        return MockResponse({"key1": "value1"}, "sample text", 404)
+    elif args[0] == "https://www.amazon.com/gp/product/B07HJXVHSS":
+        return MockResponse({"key2": "value2"}, "sample text", 200)
+
+    return MockResponse(None, 404)
+
+
+def test_boil_soup():
+    with mock.patch("requests.get", side_effect=mocked_requests_get):
```

Comment:
This is great how you are mocking these out. I would move to doing this with a decorator. This is kinda a python specific pattern, but very heavily used. So its really good to get comfortable with it. This looks like a good article (which I haven't read), but just found from a quick google search. https://medium.com/uckey/how-mock-patch-decorator-works-in-python-37acd8b78ae

Reply:
great, i watched a bunch of videos and read articles on decorators thinking i was going to use them for something else and hadn't figured out what to use them on yet